### DISCUSSION TITLE
Allow for SQLite file database alternative

### DISCUSSION THREAD

**Discussion by TheZoker** [lukevella/rallly#169] [Ideas]
Would be nice, if rallly would support SQlite file database as an alternative to Postgres database. That way the docker compose file could get rid of the postgres image and make the container smaller.

**Comment by lukevella** [comment]
Hey @TheZoker the complexity of having to support different data stores is not worth the benefits in my opinion. There's always the option to fork and modify it yourself if that's what you want. [Prisma](https://prisma.io) which is the ORM used makes it easy enough to switch data stores but there's no guarantee that will just work.

**Reply by ocdtrekkie** [reply]
Just as a note, @lukevella, I considered filing an issue for this, but since I found your note on this discussion, I will just leave a note of support for the parent post here: I think applications need a really good excuse for not just using SQLite at this point, to the extent that projects not supporting SQLite should probably include an explanation of why SQLite isn't adequate to the task. ;) This project is really neat, but I am probably not going to use it at the moment because of the hefty database requirement.

**Reply by lukevella** [reply]
Hi @ocdtrekkie, Rallly is built to handle a large number of concurrent users which is why SQLite would not be a good solution for this project. The cloud-hosted version of Rallly services many thousands of users every day. The `votes` tables is very active with well over a million rows even after daily housekeeping operations that remove inactive polls.

**Reply by ocdtrekkie** [reply]
@lukevella That's a pretty good reason. ;) Self-hosters probably wouldn't see that sort of usage, but it makes sense for a public cloud version for sure.
